framework/DataObjects/DataSet.py

```diff
@@ -1573,12 +1590,18 @@ def _getRealizationFromCollectorByValue(self, toMatch, noMatch, tol=1e-15):
     if noMatch is None:
       noMatch = {}
     assert(self._collector is not None)
+    if options:
+      allMatch = options.get("returnAllMatch",False)
```

Comment:
Dumb question, but couldn't we get all the matches by asking for noMatch = {}?

Reply:
yes, you right. this would be another way to code it.

Reply:
In that case, I would highly recommend removing the returnAllMatch option and just using noMatch = {} since it allows the same thing but without having to add a redundant feature.  Thanks.

Reply:
you are right. This makes much more sense...

Reply:
edited

Reply:
Actually I am mistaken here...allMatch means that  we are going to get all the realization that match...so if I am looking for a "groupID", I will get a list of a match realization...so it is not equivalent to noMatch = {}. I need to revert the modification

Reply:
I will find a more elegant way to do it....

Reply:
Okay, that answers my question.  Issue #1433  will fix it.